packages/manganis/manganis-core/src/folder.rs

```diff
@@ -14,7 +14,11 @@ use crate::AssetOptions;
     serde::Serialize,
     serde::Deserialize,
 )]
-pub struct FolderAssetOptions {}
+pub struct FolderAssetOptions {
+    /// If the folder's files should be optimized.
+    /// Defaults to true.
+    optimize_files: bool,
```

Comment:
Very unfortunate, but this struct was public and had no fields, meaning before this change it *could* be crafted and now it can't, meaning this technically doesn't follow semver.

We *should* have added a #[non_exhaustive] attribute on this strict.

To make this semver-safe we can add a new type that *is* non_exhaustive and use that as the FolderOptions going forward, marking the old one as deprecated. IE in the AssetOptions enum, we add a new variant that is something like "FolderLiteral" or something.

https://doc.rust-lang.org/cargo/reference/semver.html#struct-add-public-field-when-no-private

Take a look at our options if you can - we might be do something like converting FolderAssetOptions to PreservedFolder or something with a utility method.

Reply:
I want to avoid extra types and such and if this is only a temporary solution until 0.7, I'd be fine with it.

Reply:
Making it temporary is fine or forcing an opt-in with a feature flag might also work.
That way in the playground you can directly import manganis-core in the dep tree and enable the feature.

Reply:
I have gone with a new asset option type `PreservedFolderAssetOptions`. The `FolderAssetOptions` type has an `into_preserved()` method which converts it into the preserved type.

Additionally, a trait `FolderAsset` has been created so that `cli-opt` can accept either asset options for a folder asset. All the temporary structs, methods, and traits are marked `doc_hidden` to discourage use.

Let me know what you think.